pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -0,0 +1,431 @@
+        data: functionData,
+      });
+    });
+
+    it('cannot be executed twice', async () => {
+      const id = await schedule(authenticatedContract, [executor]);
+      await advanceTime(delay);
+
+      await authorizer.execute(id, { from: executor });
+      await expect(authorizer.execute(id, { from: executor })).to.be.revertedWith('ACTION_ALREADY_EXECUTED');
+    });
+
+    it('reverts if an executor is specified twice', async () => {
+      await expect(schedule(authenticatedContract, [executor, executor])).to.be.revertedWith('DUPLICATE_EXECUTORS');
+    });
+
+    it('reverts if there is no delay set', async () => {
+      action = await actionId(authenticatedContract, 'secondProtectedFunction');
+      await authorizer.grantPermission(action, user, authenticatedContract, { from: root });
+
+      await expect(
+        authorizer.instance
+          .connect(user)
+          .schedule(
+            authenticatedContract.address,
+            authenticatedContract.interface.encodeFunctionData('secondProtectedFunction', [functionData]),
+            []
+          )
+      ).to.be.revertedWith('DELAY_IS_NOT_SET');
+    });
+
+    it('reverts if the sender has permissions for another contract', async () => {
+      await expect(schedule(anotherAuthenticatedContract)).to.be.revertedWith('SENDER_DOES_NOT_HAVE_PERMISSION');
+    });
+
+    it('reverts if the sender has permissions for another action', async () => {
+      action = await actionId(authenticatedContract, 'secondProtectedFunction');
+      await authorizer.scheduleAndExecuteDelayChange(action, delay, { from: root });
+      await expect(
+        authorizer.instance
+          .connect(user)
+          .schedule(
+            authenticatedContract.address,
+            authenticatedContract.interface.encodeFunctionData('secondProtectedFunction', [functionData]),
+            []
+          )
+      ).to.be.revertedWith('SENDER_DOES_NOT_HAVE_PERMISSION');
+    });
+
+    it('reverts if the sender does not have permission', async () => {
+      await expect(
+        authorizer.instance
+          .connect(other)
+          .schedule(
+            authenticatedContract.address,
+            authenticatedContract.interface.encodeFunctionData('protectedFunction', [functionData]),
+            []
+          )
+      ).to.be.revertedWith('SENDER_DOES_NOT_HAVE_PERMISSION');
+    });
+
+    it('reverts if the target is the authorizer', async () => {
+      const where = authorizer.instance;
+      await expect(schedule(where)).to.be.revertedWith('CANNOT_SCHEDULE_AUTHORIZER_ACTIONS');
+    });
+
+    it('reverts the target is the execution helper', async () => {
+      const where = await authorizer.instance.getTimelockExecutionHelper();
+      await expect(schedule(where)).to.be.revertedWith('ATTEMPTING_EXECUTION_HELPER_REENTRANCY');
+    });
+  });
+
+  describe('execute', () => {
```

Comment:
If you're feeling spicy, we could test a scheduled execution where the scheduled action attempts to execute another (or even the same) scheduled action. This will fail because of the reentrancy guard.

Reply:
This would require another contract which calls authorizer, correct?

Reply:
Yes, it would. But now I think maybe we should remove the guard.